### ISSUE TITLE
CMake Dependency Cleaning

### ISSUE THREAD

**Issue by LeStarch** [nasa/fprime#278]
Fix the CMake dependencies internally to:

1. Unify dependencies:
    1. Static
    2. Ai detected modules
    3. AI detected files and headers
2. Way getting recursive decencies out
3. Top-level targets that can recurse

**Comment by LeStarch** [comment]
Let's revisit this with F´´ integration.  Some suggestions at that point:

1. Make the AC call a target.
2. Targets can both consume known inputs, but also produce data
   1. Targets take active ac files, but can produce more ac files
   2. Targets take in source lists, but produce new sources (enabling custom autocoders, F´´ included)
3.  Separate F´ targets from framework setup enabling pre-F´ registration and post.

In short the life-cycle should look like the following.  For **each** module:

1. Take in file and static dependency lists
2. Mine for dependencies, aka depends step
3. Iterate targets in order accumulating source files, ac files, and dependencies

Order of targets would specify build order and provide more extensibility.  Redundant items would fall away.

**Comment by LeStarch** [comment]
In v3, this is done.
